app/src/main/java/com/example/habitsmasher/AddPictureDialog.java

```diff
@@ -0,0 +1,204 @@
+                if (resultCode == RESULT_OK) {
+                    // Set selected picture
+                    _parentDialog.setImage(_selectedImage);
+
+                    // Close dialog
+                    this.dismiss();
+                }
+                break;
+        }
+    }
+
+    /**
+     * Defines the logic when the cancel button is clicked
+     */
+    protected void setCancelButtonListener() {
+        _cancelButton.setOnClickListener(new ClickListenerForCancel(getDialog(), TAG));
+    }
+
+    /**
+     * Defines the logic when the gallery button is clicked
+     */
+    protected void setGalleryButtonListener() {
+        _galleryButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Intent pickPhoto = new Intent(Intent.ACTION_PICK, MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
+                startActivityForResult(pickPhoto, 1);
+            }
+        });
+    }
+
+    /**
+     * Defines the logic when the camera button is clicked
+     */
+    protected void setCameraButtonListener() {
+        _cameraButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                dispatchTakePictureIntent();
+            }
+        });
+    }
+
+    /**
+     * Method that handles the event related to handling the camera
+     * Reference: https://developer.android.com/training/camera/photobasics#java
+     */
+    private void dispatchTakePictureIntent() {
+        Intent takePictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        // Ensure that there's a camera activity to handle the intent
+        if (takePictureIntent.resolveActivity(getContext().getPackageManager()) != null) {
+            // Create the File where the photo should go
+            File photoFile = null;
+            try {
+                photoFile = createImageFile();
+            } catch (IOException ex) {
+                Log.d(TAG, "Failed to process image");
+            }
+            // Continue only if the File was successfully created
+            if (photoFile != null) {
+                _selectedImage = FileProvider.getUriForFile(getContext(),
+                        "com.example.habitsmasher.fileprovider",
+                        photoFile);
+                takePictureIntent.putExtra(MediaStore.EXTRA_OUTPUT, _selectedImage);
+                startActivityForResult(takePictureIntent, 2);
+            }
+        }
+    }
+
+    /**
+     * Creates a temporary image file when the camera is used
+     * Reference: https://developer.android.com/training/camera/photobasics#java
+     * @return The created image
+     * @throws IOException
+     */
+    private File createImageFile() throws IOException {
+        // Create an image file name
+        String timeStamp = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());
```

Comment:
Minor: could extract the data format into a constant

Reply:
Good idea :)